projects/packages/my-jetpack/src/class-wpcom-products.php

```diff
@@ -102,8 +140,15 @@ private static function is_cache_old() {
 		if ( empty( self::get_products_from_cache() ) ) {
 			return true;
 		}
+
+		// This allows the cache to reset after the site or user connects/ disconnects
+		$check_hash = get_user_meta( get_current_user_id(), self::CACHE_CHECK_HASH_NAME, true );
+		if ( $check_hash !== self::build_check_hash() ) {
+			return true;
+		}
+
 		$cache_date = get_user_meta( get_current_user_id(), self::CACHE_DATE_META_NAME, true );
-		return time() - (int) $cache_date > ( 7 * DAY_IN_SECONDS );
+		return time() - (int) $cache_date > DAY_IN_SECONDS;
```

Comment:
Is there a reason we are reducing this from a week to a day? I am not sure why 7 days was chosen initially, but also was wondering the reason for the change

Reply:
My reasoning for reducing it was so that the pricing in My Jetpack would update sooner if the user's currency was changed for some reason. Without disconnecting, they should see new pricing in a day instead of a week. 

I think this may also help us when prices change - the product cache shouldn't hold on to old pricing for too long.